### PR TITLE
fix: add ESLint Probe for default files as well as JSON and JSONC

### LINKED ISSUE #244: 🛠 Tooling: Should set eslint.probe in ESLint settings to include .json, etc.
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

TIL that the VS Code ESLint extension will stay silent _unless_ a file in one of the following default language formats is found: `"javascript", "javascriptreact", "typescript", "typescriptreact", "html", "vue", "markdown"]`. https://github.com/microsoft/vscode-eslint/blob/560df864580a9381d84b2f1e408f46f6600fda48/README.md#settings-options

To repro:

1. Open the repo folder in VS Code without opening any files by default
2. Open any `.json` and/or `.jsonc` file(s)

VS Code will give a little prompt saying:

```plaintext
ESLint is not running. By default only TypeScript and JavaScript files are validated. If you want to validate other file types please specify them in the 'eslint.probe' setting.
```

This repo configures linting for other extensions such as `.json`, `.jsonc`, and `.md`. We should update the `eslint.probe` setting in `.vscode/settings.json` to include the corresponding VS Code language identifiers.

Note that an _extension_ (e.g. `.json`) is not the same as a _language identifier_ (e.g. `"json"`). https://code.visualstudio.com/docs/languages/identifiers 

### Additional Info

Screenshot of a VS Code prompt with the aforementioned text

### PR DESCRIPTION
**Pull request by the-lazy-learner**
## PR Checklist

- [x] Addresses an existing open issue: fixes #244
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Added `eslint.probe` to settings.json file with all the default formats alongside including .json and .jsonc.

### CODE REVIEW

**Comment by the-lazy-learner** [comment]
Please verify and suggest modifications after reviewing the changes. Thank you!

File: .vscode/settings.json
Comment (JoshuaKGoldberg) [inline .vscode/settings.json:8]: This template doesn't have tooling support for `"html"` or `"vue"` files. I think we can keep them out. There are a ton of random other languages that folks could add tooling support for on their own (CSS, Less, etc.) that this template doesn't support. What do you think?

```suggestion
	"eslint.probe": ["javascript", "javascriptreact", "typescript", "typescriptreact", "markdown", "jsonc", "json"],
```

Comment (JoshuaKGoldberg) [inline .vscode/settings.json:8]: Nitpick: could you please alphabetize this list? I know it's very nitpicky but I find it much easier to scan through lists when they're in order. 😄

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Cool, this looks good to me! Just a couple of small things please. Thanks for sending! ☺️

**Comment by JoshuaKGoldberg** [comment]
Also, it looks like the Prettier build step is failing: https://github.com/JoshuaKGoldberg/template-typescript-node-package/actions/runs/4270457190/jobs/7641323882

```plaintext
Run pnpm format --list-different

> template-typescript-node-package@1.2[7](https://github.com/JoshuaKGoldberg/template-typescript-node-package/actions/runs/4270457190/jobs/7641323882#step:4:8).5[8](https://github.com/JoshuaKGoldberg/template-typescript-node-package/actions/runs/4270457190/jobs/7641323882#step:4:9) format /home/runner/work/template-typescript-node-package/template-typescript-node-package
> prettier "**/*" --ignore-unknown "--list-different"

.vscode/settings.json
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

You'll want to run [Prettier](https://prettier.io) locally, such as with `pnpm run format:write`. Or save the file using an editor with a [Prettier extension](https://prettier.io/docs/en/editors.html) installed.

File: .vscode/settings.json
Comment (the-lazy-learner) [reply .vscode/settings.json:8]: Okay, will be making these changes in sometime

Comment (the-lazy-learner) [reply .vscode/settings.json:8]: Please have a look at it now.

**Comment by the-lazy-learner** [comment]
Please check now

File: .vscode/settings.json
Comment (JoshuaKGoldberg) [inline .vscode/settings.json:5]: There are two build failures on this file:

* An ESLint failure from the [`jsonc/sort-keys` rule](https://ota-meshi.github.io/eslint-plugin-jsonc/rules/sort-keys.html) saying `eslint.probe` should come before `eslint.rules.customizations` [[eslint failure](https://github.com/JoshuaKGoldberg/template-typescript-node-package/actions/runs/4519778310/jobs/7961845145?pr=267)]
* A Prettier failure from this file not being formatted [[prettier failure](https://github.com/JoshuaKGoldberg/template-typescript-node-package/actions/runs/4519778310/jobs/7961845145?pr=267)]

I mentioned the Prettier failure here: https://github.com/JoshuaKGoldberg/template-typescript-node-package/pull/267#issuecomment-1460886712. The [`.github/DEVELOPMENT.md` file](https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/b994ec9fe51d9ba687de70f40ecaa89edff8b0c6/.github/DEVELOPMENT.md) describes how to run those tools locally. Would you like help with this? I'm happy to either help, or just merge this PR for you if you don't have the time. Let me know! ❤️

File: .vscode/settings.json
Comment (JoshuaKGoldberg) [inline .vscode/settings.json:8]: Sorry, one more tool 😄 #285 added support for the `"yaml"` file in linting to this repo:

```suggestion
	"eslint.probe": ["javascript", "javascriptreact", "json", "jsonc", "markdown", "typescript", "typescriptreact", "yaml"],
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Getting there!

File: .vscode/settings.json
Comment (the-lazy-learner) [reply .vscode/settings.json:5]: Sure, I'll try.

File: .vscode/settings.json
Comment (the-lazy-learner) [reply .vscode/settings.json:8]: The prettier changes happened sort of because the first time I tried sending the PR I didn't clone locally, but through github.dev. Hope this fixes things! Have a happy weekend or great start to your week!

**Comment by the-lazy-learner** [comment]
Waiting for the next review. Hope you've had a good week so far. I should rethink about using github.dev sometimes.

File: .vscode/settings.json
Comment (JoshuaKGoldberg) [reply .vscode/settings.json]: Thanks, likewise on the weekend & week! ❤️

**Comment by JoshuaKGoldberg** [comment]
> github.dev

You know, I've been meaning to try using it more 😄. As long as you install the suggested extensions and run `pnpm install`, it should be good to go... but since terminals aren't supported in the web editor, I don't know how to get it to `pnpm install` or if that's doable. 🤷

**Comment by JoshuaKGoldberg** [comment]
> Waiting for the next review.

Btw I recently added docs suggesting requesting review, sorry it took a little while! #248

**Comment by JoshuaKGoldberg** [comment]
@allcontributors add @the-lazy-learner for tool

Sorry, missed this till now!

**Comment by JoshuaKGoldberg** [comment]
I'm working on a tool to do this automatically, will fill it in here soon...

**Comment by JoshuaKGoldberg** [comment]
https://github.com/JoshuaKGoldberg/template-typescript-node-package/commit/96d5f3eafa242368afcbcb58dabef4c237715a35 added it 😄

**Pull request merged**
